packages/sitecore-jss-nextjs/src/revalidate/revalidate-middleware.test.ts

```diff
@@ -204,11 +191,42 @@ describe('RevalidateMiddleware', () => {
     expect(res.revalidate).to.have.been.calledWithExactly('/About');
   });
 
-  it('should throw 500 if revalidation fails', async () => {
+  it('should return proper paths when other locales are configured besides defaultLocale', async () => {
     const res = mockResponse();
     const req = mockRequest({ ...mockUpdatedPaths });
 
+    const middleware = new RevalidateMiddleware({
+      clientFactory,
+      languages: { locales: ['en', 'fr-FR'], defaultLocale: 'en' },
+    });
+    await middleware.getHandler()(req, res);
+
+    expect(res.status).to.be.calledWith(200);
+    expect(res.json).to.be.calledWith({ revalidated: true });
+    expect(res.revalidate).to.have.been.calledWithExactly('/fr-FR/');
+    expect(res.revalidate).to.have.been.calledWithExactly('/fr-FR/About');
```

Comment:
I don't think this is right. The mock payload only has updates made to the `en` version, so we wouldn't revalidate the `fr-FR` prefixed paths. However, if the mock payload were this:
```
 {
      identifier: 'my-site/About',
      entity_definition: 'LayoutData',
      operation: 'Update',
      entity_culture: 'fr-FR',
    },
    {
      identifier: 'my-site/',
      entity_definition: 'LayoutData',
      operation: 'Update',
      entity_culture: 'fr-FR',
    },
```
then it would.

The i18n settings should only be used to determine the locale prefix of the revalidated path. Something like this flow:
- grab locale from `entity_culture`
- is locale in i18n locales?
- if no, then don't revalidate this item (skip)
- is locale the i18n defaultLocale?
- if yes, then don't add locale prefix
- if no, then add locale prefix

Of course this would only apply to Sub-path routing (our default). But what if they were using Domain routing? Perhaps we should make the locale prefix the responsibility of the app layer. Let's discuss @sc-addypathania .